stdlib/public/Concurrency/TaskGroup.cpp

```diff
@@ -703,13 +701,22 @@ static void swift_taskGroup_wait_next_throwingImpl(
   case PollStatus::Empty:
   case PollStatus::Error:
   case PollStatus::Success:
+    SWIFT_TASK_DEBUG_LOG("poll group = %p, task = %p, ready task available = %p",
+                         group, waitingTask, polled.retainedTask);
     fillGroupNextResult(context, polled);
+    if (auto completedTask = polled.retainedTask) {
+      // it would be null for PollStatus::Empty, then we don't need to release
+      group->detachChild(polled.retainedTask);
+      swift_release(polled.retainedTask);
+    }
```

Comment:
the release must of course happen AFTER the detach since the detach will inspect the task for it's next task in the invasive linked list that they form.